src/test/daml/Daml/Finance/Instrument/Swap/Test/Util.daml

```diff
@@ -209,4 +221,94 @@ lifecycleAndVerifyFinalSwapPaymentEffects readAs today swapInstrument settlers i
   let settlerUsed = head $ toList settlers
   [investorCashTransferableFixLegCid, investorCashTransferableFloatingLegCid] <- submitMulti [settlerUsed] readAs do exerciseCmd result.batchCid Batch.Settle with actors = singleton settlerUsed
 
-  pure newSwapInstrumentKey
+  pure (newSwapInstrumentKey)
+
+-- | Verify the fix payments from a payment date of a credit default swap.
+lifecycleAndVerifyCreditDefaultSwapPaymentEffectsAndSettlement : [Party] -> Date -> Instrument.K -> Parties -> Party -> Party -> ContractId Transferable.I -> ContractId Transferable.I -> AccountKey -> AccountKey -> [(Text, Parties)] -> Party -> [ContractId Observable.I] -> Script (Instrument.K, ContractId Transferable.I)
+lifecycleAndVerifyCreditDefaultSwapPaymentEffectsAndSettlement readAs today swapInstrument settlers issuer investor investorSwapTransferableCid investorCashTransferableFixLegCid custodianAccount investorAccount obs custodian observableCids = do
```

Comment:
I think we should simplify all these lifecycle tests to just verify what's in the effect being returned. It would remove a lot of the boilerplate of going through settlement, which should anyway be tested by the settlement tests. This can be done in a separate PR though, maybe just open a ticket for now.

Reply:
I agree, I have opened a ticket now: https://github.com/digital-asset/daml-finance/issues/171